src/lib/core/CHIPTLVDebug.cpp

```diff
@@ -147,7 +147,10 @@ static void DumpHandler(DumpWriter aWriter, const char * aIndent, const TLVReade
         case kTLVType_ByteString:
             err = temp.GetDataPtr(strbuf);
             VerifyOrExit(err == CHIP_NO_ERROR, aWriter("Error in kTLVType_ByteString"));
-            aWriter("%p\n", strbuf);
+            for (uint32_t i = 0; i < len; i++)
```

Comment:
@mrjerryjohns PTAL if these changes in the `CHIPTLVDebug.cpp` are valid.

Reply:
Can we have a unit test for these things?

Reply:
+1 to unit test

Reply:
Added unit test CheckOctetStringPrettyPrinter()